### PR TITLE
feat: expand Tauri config documentation page

### PR DESCRIPTION
**Pull request by lucasfernog**
adds two sections on the Tauri config documentation page:

- platform-specific config files
- extend config via the CLI

### CODE REVIEW

File: src/content/docs/develop/configuration-files.mdx [hunk 1]
```diff
@@ -71,6 +77,103 @@ endpoints = ["https://my.app.updater/{{target}}/{{current_version}}"]
...
+
+In addition to the default configuration file, Tauri can read a platform-specific configuration from:
+
+- `tauri.linux.conf.json` or `Tauri.linux.toml` for Linux
+- `tauri.windows.conf.json` or `Tauri.windows.toml` for Windows
+- `tauri.macos.conf.json` or `Tauri.macos.toml` for macOS
+- `tauri.android.conf.json` or `Tauri.android.toml` for Android
+- `tauri.ios.conf.json` or `Tauri.ios.toml` for iOS
+
+The platform-specific configuration file gets merged with the main configuration object following the [JSON Merge Patch (RFC 7396)] specification.
```
Comment (dreyfus92) [inline src/content/docs/develop/configuration-files.mdx:90]:
```suggestion
The platform-specific configuration file gets merged with the main configuration object following the [JSON Merge Patch (RFC 7396)](https://www.rfc-editor.org/info/rfc7396) specification.
```

File: src/content/docs/develop/configuration-files.mdx [hunk 2]
```diff
@@ -71,6 +77,103 @@ endpoints = ["https://my.app.updater/{{target}}/{{current_version}}"]
...
+}
+```
+
+Additionally you can provide a configuration to be merged via the CLI, see the following section for more information.
+
+### Extending the Configuration
+
+The Tauri CLI allows you to extend the Tauri configuration when running one of the `dev`, `android dev`, `ios dev`, `build`, `android build`, `ios build` or `bundle` commands.
+The configuration extension can be provided by the `--config` argument either as a raw JSON string or as a path to a JSON file.
+Tauri uses the [JSON Merge Patch (RFC 7396)] specification to merge the provided configuration value with the originally resolved configuration object.
```
Comment (dreyfus92) [inline src/content/docs/develop/configuration-files.mdx:151]:
```suggestion
Tauri uses the [JSON Merge Patch (RFC 7396)](https://www.rfc-editor.org/info/rfc7396) specification to merge the provided configuration value with the originally resolved configuration object.
```

**Review by dreyfus92 — COMMENTED** [review]
Awesome work @lucasfernog, I added a couple of suggestions, let me know what you think! 😁

File: src/content/docs/develop/configuration-files.mdx (hunk 1 above)
Comment (lucasfernog) [reply src/content/docs/develop/configuration-files.mdx:90]: didn't i include this link at the bottom of the page?

Comment (dreyfus92) [reply src/content/docs/develop/configuration-files.mdx:90]: my bad, I didn't see it.

**Review by dreyfus92 — APPROVED** [review]
LGTM ✅

**Comment by lucasfernog** [comment]
thanks for the review!

**Pull request merged**